pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -993,17 +1020,42 @@ class Timedelta(_Timedelta):
         return self.value // other.value
 
     def __rfloordiv__(self, other):
+        # numpy does not implement floordiv for timedelta64 dtype, so we cannot
+        # just defer
+        if hasattr(other, '_typ'):
+            # Series, DataFrame, ...
+            return NotImplemented
+
         if hasattr(other, 'dtype'):
-            # work with i8
-            other = other.astype('m8[ns]').astype('i8')
-            return other // self.value
+            if other.dtype.kind == 'm':
+                # also timedelta-like
+                # We need to watch out for np.timedelta64('NaT')
+                if np.ndim(other) == 0:
+                    if np.isnat(other):
+                        return np.nan
+                    else:
+                        other = other.astype('m8[ns]').astype('i8')
+                        return other // self.value
+                else:
+                    mask = np.isnat(other)
+                    res = other.astype('m8[ns]').astype('i8') // self.value
+                    if mask.any():
+                        res = res.astype('f8')
+                        res[mask] = np.nan
+                    return res
+            else:
+                raise TypeError('Invalid dtype {dtype} for '
+                                '{op}'.format(dtype=other.dtype,
+                                              op='__floordiv__'))
 
+        if isinstance(other, float) and np.isnan(other):
```

Comment:
huh? 
why are you not just checking isna

Reply:
Because `np.nan` (which can't be detected with `other is np.nan`) gets treated differently from `pd.NaT`.

Reply:
why? nan is equiv of nat here

Reply:
> why? nan is equiv of nat here

That's slightly ambiguous, but I lean towards "no it isn't".  `Timedelta(nan)` does return `NaT`, but `Timedelta / float` is meaningful on its own, so we should treat `nan` as a float for the purposes of this operation.

Reply:
checknull(other) and is_float_object(other) is more clear here (and add a comment)